### PR TITLE
feature(volumebinding): return Skip in PreFilter

### PR DESCRIPTION
**Pull request by AxeZhan**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
return Skip in volumebinding PreFilter when the following condition is satisfied:

- pod doesn't have a pvc.

#### Which issue(s) this PR fixes:

Part of #114399

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by sanposhiho — COMMENTED** [review]
/ok-to-test

**Review by sanposhiho — APPROVED** [review]
/lgtm

Looking good to me, thanks @kidddddddddddddddddddddd

File: pkg/scheduler/framework/plugins/volumebinding/volume_binding.go [hunk 1]
```diff
@@ -166,8 +165,8 @@ func (pl *VolumeBinding) PreFilter(ctx context.Context, state *framework.CycleSt
 	if hasPVC, err := pl.podHasPVCs(pod); err != nil {
 		return nil, framework.NewStatus(framework.UnschedulableAndUnresolvable, err.Error())
 	} else if !hasPVC {
-		state.Write(stateKey, &stateData{skip: true})
-		return nil, nil
+		state.Write(stateKey, &stateData{})
```
Comment (sanposhiho) [inline pkg/scheduler/framework/plugins/volumebinding/volume_binding.go:168]: Why do we need to write state when returning Skip?

Comment (kidddddddddddddddddddddd) [reply pkg/scheduler/framework/plugins/volumebinding/volume_binding.go:168]: We'll try to getStateData in `Reserve`, if we don't write state here, reserve will treat no state data as a no founding error.
[logic](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/volumebinding/volume_binding.go#L308)

Comment (sanposhiho) [reply pkg/scheduler/framework/plugins/volumebinding/volume_binding.go:168]: Ah, I see.
Hmm, I'd say we can just ignore the error from `getStateData` in `Reserve` when [cyclestate.SkipFilterPlugins](https://github.com/kubernetes/kubernetes/blob/3aff1f97bef13f1d40e47fae785994d57c3e5a80/pkg/scheduler/framework/cycle_state.go#L54) has `VolumeBinding` plugin.

Comment (kidddddddddddddddddddddd) [reply pkg/scheduler/framework/plugins/volumebinding/volume_binding.go:168]: Here are two concerns:

1.  If we just ignore the error from `getStateData` in Reserve,  we'll still need to constrcut a `StateData` with `allBound` field and write it back to the cycleState. [logic](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/volumebinding/volume_binding.go#L319). So I don't think there's too much difference.
2. `getStateData` is used in many stages:`Filter`,`PreBind`,`Reserve`,`Score`,`Unreserve`.If we ignore this err in `Reserve`, we'll have to check `SkipFilterPlugins` in other stages too. There will be more code changes involved.

WDYT?

Comment (sanposhiho) [reply pkg/scheduler/framework/plugins/volumebinding/volume_binding.go:168]: > getStateData is used in many stages:Filter,PreBind,Reserve,Score,Unreserve.If we ignore this err in Reserve, we'll have to check SkipFilterPlugins in other stages too.

That makes sense. Ok, let's just keep it as it is.

**Review by sanposhiho — APPROVED** [review]
Thanks @kidddddddddddddddddddddd :)

/lgtm

**Comment by kidddddddddddddddddddddd** [comment]
/assign @gnufied 
for another review and approve

**Pull request merged**